### DISCUSSION TITLE
Change Font size and dropdown width

### DISCUSSION THREAD

**Discussion by samitray** [PhilippeMarcMeyer/vanillaSelectBox#44] [Q&A] [answered]
Hi,

Is there any way to change the font size of the option label, I can see in the demo page how to change the font color.

Is there a setting to change the dropdown size, my select box size is small, but the drop down width is more than 
the select box. or can the drop down width respect the minwidth /maxwidth

**Comment by PhilippeMarcMeyer** [comment, marked as answer]
That not easy 
we could reduce the width of the menu
but at the present time, VSB css is not designed for several font-size
the check marks are dependant upon the font size and would be misplaced
that something i should deal with in the future

**Reply by samitray** [reply]
Agreed, check box and font size do go together.
Option dropdown width specially for small select boxes having option labels of 2-3 chars.
If the user make the width very small then display partial label and ..... (ellipsis)

**Reply by PhilippeMarcMeyer** [reply]
yes that could be a start ! Ill do it as soon as possible. For the future, the css limitations should be adressed one day. There has been demands for width limitations but also night mode or other check styles !
the limitation concerning the check marks is that it is just a square with only 2 borders and rotated by 45° ! maybe i could use awesome fonts instead but then there would be dependency...

**Reply by PhilippeMarcMeyer** [reply]
Here it is : I have added an new option in v 0.62 : maxOptionWidth : set a maximum width for each option for narrow menus

let selectCars = new vanillaSelectBox("#demoShort",{"maxOptionWidth":70, "maxHeight": 200,"search": true,translations: { "all": "All", "items": "Cars" } });

**Reply by samitray** [reply]
font awsome would create dependency and add a whole lot of size, however if a svg with rounded rect and a tick path that can be set visible or hidden is extremely light and svg can be very easily scaled to match font size, will try to generate sample svg and post back in a couple of days
